### DISCUSSION TITLE
Setting environment variables in directives

### DISCUSSION THREAD

**Discussion by bishabosha** [VirtusLab/scala-cli#1815] [Ideas]
it seems useful to be able to declare environment variables in the directives, so a launch script wrapper is not needed

**Comment by tgodzik** [comment]
What kind of variables would you need to set, that would be needed in the script? Are there not ways of setting them inside the script or supplying the parameters via different means?

**Reply by bishabosha** [reply]
The use case I imagined would be launching from a code lens in VS Code. I know that sometimes environment flags are required by linked native libraries.

**Reply by tgodzik** [reply]
You can always launch code from the console with those variables set, or have them set globally if needed. Though, having Scala native working with code lenses is still a long way to go :sweat_smile: And also this would probably use the build tool, so you would most likely be able to use the `file.env` to set everything up (this is already supported for debug)
